### PR TITLE
Feature/72 map implementation

### PR DESCRIPTION
**Pull request by Arthelh**

### CODE REVIEW

File: app/src/main/java/com/ncnf/map/MapHandler.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import com.mancj.materialsearchbar.MaterialSearchBar;
+import com.mancj.materialsearchbar.adapter.SuggestionsAdapter;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static android.content.Context.INPUT_METHOD_SERVICE;
+
+public class MapHandler {
```
Comment (Jnf14) [inline app/src/main/java/com/ncnf/map/MapHandler.java:33]: Nice to have an extra class to handle the map !

File: app/src/main/java/com/ncnf/map/MapHandler.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        }
+    }
+
+    public void setZoomLevel(float zoomLevel) {
+        this.zoomLevel = zoomLevel;
+    }
+
+    //Removes all markers from the map and recreates them according to current position
+    public void update_markers(){
+        for (Marker m : eventMarkers)
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/map/MapHandler.java:234]: single for fori loop?

File: app/src/main/java/com/ncnf/map/MapHandler.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            @Override
+            public void OnItemDeleteListener(int position, View v) {
+
+            }
+        });
+    }
+
+    public void switchMarkers(GoogleMap mMap) {
+        eventsShown = !eventsShown;
+        for (Marker e : eventMarkers) {
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/map/MapHandler.java:186]: single fori loop?

**Review by HyPn0TiiK — APPROVED** [review]

File: app/src/main/java/com/ncnf/map/MapHandler.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                    }
+                });
+            }
+
+            @Override
+            public void OnItemDeleteListener(int position, View v) {
+
+            }
+        });
+    }
```
Comment (Jnf14) [inline app/src/main/java/com/ncnf/map/MapHandler.java:182]: @Arthelh What do you think about creating a SearchBarHandler which contains these 3 functions in order to separate the Map from the SearchBar elements and modularize ? This Search bar may be useful some place else, in the settings for example to set the position I'm at once for all, etc... Tell me if you think it's a good idea !

**Review by Jnf14 — APPROVED** [review]

File: app/src/main/java/com/ncnf/map/MapHandler.java (hunk 2 above)
Comment (Arthelh) [reply app/src/main/java/com/ncnf/map/MapHandler.java:234]: Not the same range, eventMarkers and venueMarkers do not have the same size

File: app/src/main/java/com/ncnf/map/MapHandler.java (hunk 3 above)
Comment (Arthelh) [reply app/src/main/java/com/ncnf/map/MapHandler.java:186]: Not the same range, eventMarkers and venueMarkers do not have the same size nor the same markers

File: app/src/main/java/com/ncnf/map/MapHandler.java (hunk 1 above)
Comment (Arthelh) [reply app/src/main/java/com/ncnf/map/MapHandler.java:33]: Did as you said ;)

File: app/src/main/java/com/ncnf/map/MapHandler.java (hunk 4 above)
Comment (Arthelh) [reply app/src/main/java/com/ncnf/map/MapHandler.java:182]: That's definitely something to think about. We can do it in a later branch. I was also thinking once the feedView would be completely implemented, we could use the searchBar to filter events by tags or other too!

Comment (Jnf14) [reply app/src/main/java/com/ncnf/map/MapHandler.java:182]: Yep that's my point, I'll modularize that in my branch which will use fragments for most of the activities.

**Pull request merged**